posts/inside-rust/2024-08-15-this-development-cycle-in-cargo-1.81.md

```diff
@@ -0,0 +1,426 @@
+- How free should cargo be to updating other dependencies to allow a breaking update to occur?  Currently, `--precise` only allows the dependency you specified to be updated unless you pass `--recursive`.  Its likely that a breaking change will have a cascading effect.
+
+Some of these are being worked out on those PRs while others are being left to the [tracking issue](https://github.com/rust-lang/cargo/issues/12425#issuecomment-2186198258).
+
+##### `cargo publish --workspace`
+
+*Update from [1.80](https://blog.rust-lang.org/inside-rust/2024/06/19/this-development-cycle-in-cargo-1.80.html#cargo-publish---workspace)*
+
+[jneem](https://github.com/jneem) continued work on `cargo package --workspace`.
+The first step was to switch `cargo package` to run in stages, first packaging the `.crate` files and then verifying them
+([#14074](https://github.com/rust-lang/cargo/pull/14074)).
+Once that was in place, they added building multiple `.crate` files
+([#13947](https://github.com/rust-lang/cargo/pull/13947)).
+
+##### Fingerprinting builds
+
+[Xaeroxe](https://github.com/Xaeroxe) restarted the conversation around fingerprinting source using checksums, instead of mtimes, on
+[zulip](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Implementing.20checksum.20based.20fingerprinting).
+Specifically, they were starting work on this and wanted to avoid race conditions between when cargo checksummed the source and when rustc built.
+Ideally, rustc would checksum the source to avoid this.
+Some past work in this direction was shared.
+From there, they created a compiler [MCP](https://github.com/rust-lang/compiler-team/issues/765)
+and a [PR](https://github.com/rust-lang/rust/pull/126930) against the compiler.
+However, to do this with `build.rs` would require `build.rs` files to opt-in and they would need to be coordinate on checksum algorithms.
+We decided to punt on this and leave `build.rs` using mtime for now.
+
+[Xaeroxe](https://github.com/Xaeroxe) posted [#14137](https://github.com/rust-lang/cargo/pull/14137) an initial version of checksum fingerprinting for Cargo.
+
+##### `cargo info`
+
+*Update from [1.79](https://blog.rust-lang.org/inside-rust/2024/05/07/this-development-cycle-in-cargo-1.79.html#cargo-info)*
+
+After waiting some time to collect feedback,
+[Rustin170506](https://github.com/Rustin170506)
+proposed `cargo info` to be merged into cargo ([#14141](https://github.com/rust-lang/cargo/pull/14141)).
+This included adapting the code from being written as a third-party to being written in a way Cargo expects, documentation was added, and completion scripts.
+
+There was some discussion on whether to insta-stabilize.
+As a third-party command, we are able to get the testing benefits of nightly-only features.
+If we don't insta-stabilize it, you could only run it as `cargo +nightly -Zsubcommand info` or run the existing `cargo-info` as `cargo info` but `cargo-info info`; `cargo info` would fail.
```

Comment:
Possible typo here, or maybe I'm just struggling to understand the sentence structure.

Maybe this was intended instead?
```suggestion
If we don't insta-stabilize it, you could only run it as `cargo +nightly -Zsubcommand info` or run the existing `cargo-info` not as `cargo info` but `cargo-info info`; `cargo info` would fail.
```